File: Recursive-Bubble-Sort.js
class Solution {
    bubbleSort(nums) {
        let n=nums.length 
        function b_s1(nums,n){
            if (n===1){
                return;
            }
            b_sp(nums,0,n)
            b_s1(nums,n-1)
        }
        function b_sp(nums,i,n){
            if (i === (n-1)){
                return 
            }
            if (nums [i]>nums[i+1]){
                [nums[i],nums[i+1]] = [nums[i+1], nums[i]]
            }
            b_sp(nums,i+1,n)
        }
        b_s1(nums,n)
        return nums
    }
}

let nums=[1,5,44,3,0,22]
console.log(new Solution().bubbleSort(nums))

File: Recursive-Bubble-Sort.py
class Solution:
    def bubbleSort(self, nums):
        n=len(nums)
        def bubbleSort1( nums, n):
            if n==1:
               return 
            bubbleSort_pass(nums,0,n)
            bubbleSort1(nums,n-1)
        def bubbleSort_pass( nums,i,n):
            if i==(n-1):
                return 
            if nums[i]>nums[i+1]:
                nums[i],nums[i+1]=nums[i+1],nums[i]
            bubbleSort_pass(nums,i+1,n)
        bubbleSort1(nums,n)
        return nums
        
arr = [5, 1, 4, 2, 8]
print(Solution().bubbleSort(arr))
